packages/utils/src/syncpromise.ts

```diff
@@ -18,7 +18,7 @@ const enum States {
  * Thenable class that behaves like a Promise and follows it's interface
  * but is not async internally
  */
-class SyncPromise<T> implements PromiseLike<T> {
+export class SyncPromise<T> implements PromiseLike<T> {
```

Comment:
I wonder if we can remove all sync promise usage in browser environments and only use it for node.